storage/src/vespa/storage/bucketdb/btree_lockable_map.hpp

```diff
@@ -179,6 +179,12 @@ void BTreeLockableMap<T>::acquireKey(const LockId& lid, std::unique_lock<std::mu
     }
 }
 
+template <typename T>
+bool BTreeLockableMap<T>::hasKey(const key_type& key) const {
+    auto readGuard = do_acquire_read_guard();
+    return _impl->find(key).valid();
```

Comment:
That find method is only supposed to be used by the writer and is not protected by the read guard.

Reply:
That was what I suspected. new wiring added.